### ISSUE TITLE
Cannot read Feather V2 file; works with pandas, vaex, pyarrow: `PanicException: called Result::unwrap() on an Err value: CastError("cannot cast list to large-list with different child data")`

### ISSUE THREAD

**Issue by tsoernes** [pola-rs/polars#784]
#### Are you using Python or Rust?
Python 3.8.8

#### What version of polars are you using?

polars 0.8.4
pyarrow 4.0.0

#### What operating system are you using polars on?

Fedora 34 x64

#### Describe your bug.

Unable to read a Feather V2 file, `lz4` compressed, written in Pandas. The file has two columns: `int` and `List[str]`.

Here is the file: https://we.tl/t-6i7E64TFpf

I have tried 3 different methods: 

1. Read file into an Arrow table with pyarrow (that works), then attempt to move into polars with `from_arrow` (does not work)
2. `read_ipc` with `use_pyarrow=True`
3. `read_ipc` with `use_pyarrow=False`

```python

"""
Inputs are a bit out of order due to readability
"""

[ins] In [8]: import pandas; import vaex; import polars; import pyarrow
[ins] In [10]: path
Out[10]: Path('/home/ftdb/data/tags/job_tag.feather.000')

"""
Reading the file works well in Pandas, Vaex and in PyArrow
"""

[ins] In [11]: df1 = pandas.read_feather(path)
[ins] In [13]: table = pyarrow.feather.read_table(path)
[ins] In [14]: df2 = vaex.from_arrow_table(table)
[ins] In [15]: len(df1)
Out[15]: 126146
[ins] In [16]: len(df2)
Out[16]: 126146
[ins] In [32]: df4 = pyarrow.feather.read_feather(path)
[ins] In [33]: len(df4)
Out[33]: 126146

"""
Now try to read it with polars, 3 different ways, and it fails
"""

[ins] In [17]: df3 = polars.from_arrow(table)
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: CastError("cannot cast list to large-list with different child data")', /github/workspace/polars/polars-core/src/series/mod.rs:1583:26
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
File "<ipython-input-17-51325b874ace>", line 1, in <module>
    df3 = polars.from_arrow(table)
    ..................................................
     polars.from_arrow = <function 'from_arrow' functions.py:378>
     table = pyarrow.Table
             job.listingId: int64
             tags: list<item: string>
               child 0, item: string
    ..................................................

File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/functions.py", line 390, in from_arrow
    378  def from_arrow(a: "Union[pa.Table, pa.Array]", rechunk: bool = True) -> "DataFrame":
 (...)
    386      rechunk
    387          Make sure that all data is contiguous.
    388      """
    389      if isinstance(a, pa.Table):
--> 390          return DataFrame.from_arrow(a, rechunk)
    391      if isinstance(a, pa.Array):
    ..................................................
     a = pyarrow.Table
         job.listingId: int64
         tags: list<item: string>
           child 0, item: string
     rechunk = True
     pa.Table = <class 'pyarrow.lib.Table'>
     pa.Array = <class 'pyarrow.lib.Array'>
    ..................................................

File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/frame.py", line 328, in from_arrow
    301  def from_arrow(table: pa.Table, rechunk: bool = True) -> "DataFrame":
 (...)
    324  
    325      table = pa.table(data)
    326      batches = table.to_batches()
    327      self = DataFrame.__new__(DataFrame)
--> 328      self._df = PyDataFrame.from_arrow_record_batches(batches)
    329      if rechunk:
    ..................................................
     table = pyarrow.Table
             job.listingId: int64
             tags: list<item: string>
               child 0, item: string
     pa.Table = <class 'pyarrow.lib.Table'>
     rechunk = True
     data = {'job.listingId': <pyarrow.lib.Int64Array object at 0x7efd73
             177ee0>
                              [
                                4001522600,
                                4001523105,
                                4001523172,
                                4001524355,
                                4001524359,
                                4001524361,
                                4001524369,
                                4001524375,
                                4001524381,
                                4001524382,
                                ...
                                4029510979,
                                4029510992,...
     batches = [
                pyarrow.RecordBatch
                job.listingId: int64
                tags: list<item: string>
                  child 0, item: string, ]
     self = # error calling repr and str
     DataFrame = <class 'polars.frame.DataFrame'>
     self._df = # AttributeError
          self = # error calling repr and str
    ..................................................

---- (full traceback above) ----
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
File "<ipython-input-17-51325b874ace>", line 1, in <module>
    df3 = polars.from_arrow(table)
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/functions.py", line 390, in from_arrow
    return DataFrame.from_arrow(a, rechunk)
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/frame.py", line 328, in from_arrow
    self._df = PyDataFrame.from_arrow_record_batches(batches)

PanicException: called `Result::unwrap()` on an `Err` value: CastError("cannot cast list to large-list with different child data")
[ins] In [19]: df3 = polars.read_ipc(path)
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: CastError("cannot cast list to large-list with different child data")', /github/workspace/polars/polars-core/src/series/mod.rs:1583:26
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
File "<ipython-input-19-5367f98fa2f8>", line 1, in <module>
    df3 = polars.read_ipc(path)
    ..................................................
     polars.read_ipc = <function 'read_ipc' functions.py:287>
     path = Path('/home/ftdb/data/tags/job_tag.feather.000')
    ..................................................

File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/functions.py", line 303, in read_ipc
    287  def read_ipc(file: Union[str, BinaryIO, Path], use_pyarrow: bool = True) -> "DataFrame":
 (...)
    299      -------
    300      DataFrame
    301      """
    302      file = _prepare_file_arg(file)
--> 303      return DataFrame.read_ipc(file, use_pyarrow)
    ..................................................
     file = Path('/home/ftdb/data/tags/job_tag.feather.000')
     Union = typing.Union
     BinaryIO = <class 'typing.BinaryIO'>
     Path = <class 'pathlib.Path'>
     use_pyarrow = True
    ..................................................

File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/frame.py", line 278, in read_ipc
    261  def read_ipc(file: Union[str, BinaryIO], use_pyarrow: bool = True) -> "DataFrame":
 (...)
    274      DataFrame
    275      """
    276      if use_pyarrow:
    277          tbl = pa.feather.read_table(file)
--> 278          return DataFrame.from_arrow(tbl)
    279  
    ..................................................
     file = Path('/home/ftdb/data/tags/job_tag.feather.000')
     Union = typing.Union
     BinaryIO = <class 'typing.BinaryIO'>
     use_pyarrow = True
     tbl = pyarrow.Table
           job.listingId: int64
           tags: list<item: string>
             child 0, item: string
     pa.feather.read_table = <function 'read_table' feather.py:220>
    ..................................................

File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/frame.py", line 328, in from_arrow
    301  def from_arrow(table: pa.Table, rechunk: bool = True) -> "DataFrame":
 (...)
    324  
    325      table = pa.table(data)
    326      batches = table.to_batches()
    327      self = DataFrame.__new__(DataFrame)
--> 328      self._df = PyDataFrame.from_arrow_record_batches(batches)
    329      if rechunk:
    ..................................................
     table = pyarrow.Table
             job.listingId: int64
             tags: list<item: string>
               child 0, item: string
     pa.Table = <class 'pyarrow.lib.Table'>
     rechunk = True
     data = {'job.listingId': <pyarrow.lib.Int64Array object at 0x7efd73
             17eca0>
                              [
                                4001522600,
                                4001523105,
                                4001523172,
                                4001524355,
                                4001524359,
                                4001524361,
                                4001524369,
                                4001524375,
                                4001524381,
                                4001524382,
                                ...
                                4029510979,
                                4029510992,...
     batches = [
                pyarrow.RecordBatch
                job.listingId: int64
                tags: list<item: string>
                  child 0, item: string, ]
     self = # error calling repr and str
     DataFrame = <class 'polars.frame.DataFrame'>
     self._df = # AttributeError
          self = # error calling repr and str
    ..................................................

---- (full traceback above) ----
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
File "<ipython-input-19-5367f98fa2f8>", line 1, in <module>
    df3 = polars.read_ipc(path)
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/functions.py", line 303, in read_ipc
    return DataFrame.read_ipc(file, use_pyarrow)
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/frame.py", line 278, in read_ipc
    return DataFrame.from_arrow(tbl)
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/frame.py", line 328, in from_arrow
    self._df = PyDataFrame.from_arrow_record_batches(batches)

PanicException: called `Result::unwrap()` on an `Err` value: CastError("cannot cast list to large-list with different child data")

[ins] In [24]: df3 = polars.read_ipc(path, False)
File "/home/ftdb/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
File "<ipython-input-24-cfb6234ba6aa>", line 1, in <module>
    df3 = polars.read_ipc(path, False)
    ..................................................
     polars.read_ipc = <function 'read_ipc' functions.py:287>
     path = Path('/home/ftdb/data/tags/job_tag.feather.000')
    ..................................................

File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/functions.py", line 303, in read_ipc
    287  def read_ipc(file: Union[str, BinaryIO, Path], use_pyarrow: bool = True) -> "DataFrame":
 (...)
    299      -------
    300      DataFrame
    301      """
    302      file = _prepare_file_arg(file)
--> 303      return DataFrame.read_ipc(file, use_pyarrow)
    ..................................................
     file = Path('/home/ftdb/data/tags/job_tag.feather.000')
     Union = typing.Union
     BinaryIO = <class 'typing.BinaryIO'>
     Path = <class 'pathlib.Path'>
     use_pyarrow = False
    ..................................................

File "/home/ftdb/anaconda3/lib/python3.8/site-packages/polars/frame.py", line 281, in read_ipc
    261  def read_ipc(file: Union[str, BinaryIO], use_pyarrow: bool = True) -> "DataFrame":
 (...)
    277          tbl = pa.feather.read_table(file)
    278          return DataFrame.from_arrow(tbl)
    279  
    280      self = DataFrame.__new__(DataFrame)
--> 281      self._df = PyDataFrame.read_ipc(file)
    282      return self
    ..................................................
     file = Path('/home/ftdb/data/tags/job_tag.feather.000')
     Union = typing.Union
     BinaryIO = <class 'typing.BinaryIO'>
     use_pyarrow = False
     pa.feather.read_table = <function 'read_table' feather.py:220>
     self = # error calling repr and str
     DataFrame = <class 'polars.frame.DataFrame'>
     self._df = # AttributeError
          self = # error calling repr and str
    ..................................................

TypeError: Object does not have a .read() method.


```

**Comment by ritchie46** [comment]
Thanks for the issue report. It is fixed in #786

**Comment by ritchie46** [comment]
Fixed and released :+1:
